src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1337,16 +1423,70 @@ int computeIndexingBin() {
         return GenomicIndexUtil.reg2bin(alignmentStart, alignmentEnd);
     }
 
+    /**
+     *
+     * @return the SAMFileHeader for this record. If the header is null, the following SAMRecord methods may throw
+     * IllegalArgumentExceptions:
+     *
+     *   getReferenceIndex
+     *   setReferenceIndex
+     *   getMateReferenceIndex
+     *   setMateReferenceIndex
+     *
+     * A SAMRecord with a null header can not be fully validated and will result in SAMValidationErrors if isValid is
+     * called.
```

Comment:
Can you copy these warnings on `isValid()` and methods that can throw exceptions into a new paragraph on "Notes on headerless SAMRecords" in the class-level comment at the top of `SAMRecord`?